src/whylogs/app/writers.py

```diff
@@ -251,10 +251,9 @@ def _write_protobuf(self, profile: DatasetProfile, rotation_suffix: Optional[str
         """
         path = self.ensure_path(os.path.join(self.path_suffix(profile), "protobuf"))
 
-        protobuf: Message = profile.to_protobuf()
-
         with open(os.path.join(path, self.file_name(profile, ".bin", rotation_suffix)), "wb") as f:
-            f.write(protobuf.SerializeToString())
+            f.write(profile.serialize_delimited())
```

Comment:
how about replace both lines above with `profile.write_protobuf(os.path.join(path, self.file_name(profile, ".bin", rotation_suffix))`

Reply:
DatasetProfile.write_protobuf() is implemented using msg.SerializeToString, which is what we're trying to avoid.
We could go ahead and replace all uses of SerializeToString with  routines with serialize_delimited - maybe that's what you're suggesting?

Reply:
https://github.com/whylabs/whylogs/blob/mainline/src/whylogs/core/datasetprofile.py#L550
`DatasetProfile.write_protobuf()` writes delimited by default FYI:

```
    def write_protobuf(self, protobuf_path: str, delimited_file: bool = True):
```

Reply:
@andyndang we will push as is for now and change the the open in Datasetprofile to a smart open so we can pass transport param to it in seperate PR.